samples/matter/thermostat/src/app_task.cpp

```diff
@@ -5,43 +5,25 @@
  */
 
 #include "app_task.h"
-#include "fabric_table_delegate.h"
+
+#include "matter_init.h"
 #include "task_executor.h"
 
 #include "temp_sensor_manager.h"
 #include "temperature_manager.h"
 
-#include <platform/CHIPDeviceLayer.h>
-
 #include <app-common/zap-generated/attributes/Accessors.h>
-#include <app/DeferredAttributePersistenceProvider.h>
 #include <app/clusters/identify-server/identify-server.h>
 #include <app/server/OnboardingCodesUtil.h>
-#include <app/server/Server.h>
-#include <credentials/DeviceAttestationCredsProvider.h>
-#include <credentials/examples/DeviceAttestationCredsExample.h>
-#include <lib/support/CHIPMem.h>
-#include <lib/support/CodeUtils.h>
-#include <system/SystemError.h>
-
-#ifdef CONFIG_CHIP_WIFI
-#include <app/clusters/network-commissioning/network-commissioning.h>
-#include <platform/nrfconnect/wifi/NrfWiFiDriver.h>
-#endif
 
 #ifdef CONFIG_CHIP_OTA_REQUESTOR
 #include "ota_util.h"
 #endif
 
-#include <dk_buttons_and_leds.h>
-#include <zephyr/kernel.h>
-#include <zephyr/logging/log.h>
```

Comment:
Logging is still used in these samples and so are button constants AFAICS

Reply:
dk_buttons_and_leds is a part of `board` module now. Without logging it compiles but I agree we cannot depend on other modules to include it for us.